columnflow/tasks/histograms.py

```diff
@@ -191,7 +191,7 @@ def run(self):
                 )
 
                 # build the full event weight
-                weight = (
+                events, weight = (
                     ak.Array(np.ones(len(events), dtype=np.float32))
                     if self.weight_producer_inst.skip_func()
                     else self.weight_producer_inst(events)
```

Comment:
Looks like the case where `skip_func()` is true needs to be adapted as well. Maybe something like the following:
```suggestion
                if not self.weight_producer_inst.skip_func():
                    events, weight = self.weight_producer_inst(events)
                else:
                    weight = ak.Array(np.ones(len(events), dtype=np.float32))
```